hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -4354,22 +4257,25 @@ public void close() throws IOException {
   protected synchronized void stopAllServices() {
     try {
       trackDuration(getDurationTrackerFactory(), FILESYSTEM_CLOSE.getSymbol(), () -> {
-        closeAutocloseables(LOG, store);
+        closeAutocloseables(LOG, getStore());
```

Comment:
@steveloughran can you explain the lifecycle of things with the Service stuff?

We're doing `closeAutocloseables(LOG, getStore());` here, but Store doesn't actually implement AutoCloseable. So I'm not sure if we need to something here (call serviceStop()?)

Also need to figure out what changes I need to make to close our s3SeekableInputStreamFactory, prior to this changes, I was passing in our AAL factory, in this `closeAutocloseables()`, which would call `.s3SeekableInputStreamFactory.close()`.